### PR TITLE
Update LLVM to e899641df2391179e8ec29ca14c53b09ae7ce85c

### PR DESCRIPTION
**Pull request by philass**

### CODE REVIEW

**Comment by hamptonm1** [comment]
@philass For the `Stablehlo` commit... we want to pull in the exact commit which maps to their LLVM upgrade. So in this case it would be `115941b8b0fc3cd289f741d6734a0040361ae725` if you are using LLVM commit `e899641df2391179e8ec29ca14c53b09ae7ce85c`. As seen from this PR in the community: https://github.com/openxla/stablehlo/commit/115941b8b0fc3cd289f741d6734a0040361ae725/

**Comment by philass** [comment]
@hamptonm1 

> For the Stablehlo I would try to map the LLVM commit as closely to the Stablehlo commit to avoid incurring more errors 

That is what I did https://github.com/openxla/stablehlo/pull/2035

**Comment by hamptonm1** [comment]
@philass The Stablehlo commit would be `[115941b](https://github.com/openxla/stablehlo/commit/115941b8b0fc3cd289f741d6734a0040361ae725)` based on what you sent me not what you currently have checked out in this PR. That is what I was implying.

**Comment by brnorris03** [comment]
Thank you for working on this update, it would be really good to have. Could you use any help with anything?

**Comment by philass** [comment]
@brnorris03 sorry I didn't realize some people were blocked by this.

I'm continuing work on this today.

Right now I'm blocked on two things
- A single StableHLO lit test that is failing when run onnx Linux (works on my Mac and the Mac CI pipeline)
- Windows build of StableHLO

I'll dig more into the two issues and keep the community posted if I get blocked

File: src/Compiler/CompilerUtils.cpp [hunk 1]
```diff
@@ -210,7 +210,7 @@ static void loadMLIR(std::string inputFilename, mlir::MLIRContext &context,
     FunctionType newType =
         FunctionType::get(&context, newArgTypes, funcType.getResults());
     ConversionPatternRewriter rewriter(&context);
```
Comment (mgehre-amd) [inline src/Compiler/CompilerUtils.cpp:212]: Is the rewriter now unused?

Comment (philass) [reply src/Compiler/CompilerUtils.cpp:212]: Good catch

**Comment by philass** [comment]
Hey @brnorris03,

I'm down to just the argmax issue.

I'm struggling to reproduce this issue locally. I've tried a few different variations of `ArgMax.cpp` that pass locally and in the mac CI pipeline. 

Are you familiar with the StableHLO backend? If you have any spare cycles and could take a look at that test it would be greatly appreciated! If not no worries

**Comment by brnorris03** [comment]
> Are you familiar with the StableHLO backend? If you have any spare cycles and could take a look at that test it would be greatly appreciated! If not no worries

I'm somewhat familiar, happy to take a look!

**Comment by brnorris03** [comment]
@philass please try this [argmax.patch](https://github.com/onnx/onnx-mlir/files/14623846/argmax.patch) for the argmax problem.

**Comment by brnorris03** [comment]
I just changed the [argmax patch](https://github.com/onnx/onnx-mlir/files/14630570/argmax.patch) slightly to use ranked tensors since that will be needed in the next upstream update.

**Comment by philass** [comment]
@brnorris03 Thanks so much! It seemed to do the trick.

I'll be happy to merge this in, and then review your follow up PR

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, thanks for this update. Looks like there were tricky updates, esp with respect to memory.

If you still have a link handy, please add a reference to the MLIR PR that describe the changes to memory interface, aka `DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]`

Big thanks again

**Pull request merged**